src/lib.rs

```diff
@@ -1093,6 +1106,43 @@ impl<K, V, S> Drop for FlurryHashMap<K, V, S> {
     }
 }
 
+impl<K, V, S> Extend<(K, V)> for &FlurryHashMap<K, V, S>
+where
+    K: Sync + Send + Clone + Hash + Eq,
+    V: Sync + Send,
+    S: BuildHasher,
+{
+    // TODO: Implement Java's `tryPresize` method to pre-allocate space for
+    // the incoming entries
+    fn extend<T: IntoIterator<Item = (K, V)>>(&mut self, iter: T) {
+        let guard = crossbeam::epoch::pin();
+        (*self).put_all(iter.into_iter(), &guard);
+    }
+}
+
+impl<K, V> FromIterator<(K, V)> for FlurryHashMap<K, V, RandomState>
+where
+    K: Sync + Send + Clone + Hash + Eq,
+    V: Sync + Send,
+{
+    fn from_iter<T: IntoIterator<Item = (K, V)>>(iter: T) -> Self {
+        let mut it = iter.into_iter();
+        let guard = crossbeam::epoch::pin();
+
+        let map = if let Some((key, value)) = it.next() {
```

Comment:
You could probably avoid some indentation here with:
```rust
let first = it.next();
if first.is_none() {
    return Self::new();
}
let Some((key, value)) = first.unwrap();
// ...
```
But not sure whether it's worth it. Up to you :)

Reply:
Ah, yes, what you did was nicer!

Reply:
Yep I could, but idk. I like to avoid using `unwrap` whenever possible, and since the code isn't indented that much I don't mind.